src/StdIOCallback.cpp

```diff
@@ -146,6 +138,14 @@ void StdIOCallback::setFilePointer(int64 Offset,seek_mode Mode)
         mCurrentPosition = Offset;
         break;
     }
+  } else {
+#if !defined(__GNUC__) || (__GNUC__ > 2)
```

Comment:
This is a weird warning from the compiler. Why would that be an error. The code does literally the same thing. Is it a readability thing ?

Also, side note, we likely don't support GCC <= 2 so we could drop the `#else` part and make it more readable.

Reply:
No idea.

Reply:
We definitely do not support gcc that old. We require C++11 support, and that was introduced [in gcc 4.8](https://en.cppreference.com/w/cpp/compiler_support) (well, the earliest features started to arrive in 4.3 already, but full support was roughly 4.8).

I noticed this as well but didn't want to request "remove support for gcc < 4.8" as not to extend the scope of this PR too much. I fully intend to remove said code soon(ish) myself if no one else does.

Reply:
We should probably create a Milestone on Github or something for this new libebml (and libmatroska) API breakage ? This way we can tell which things we have to merge, which can be merged right away and so we don't miss any when we do this breakage.